core/rint/src/TTabCom.cxx

```diff
@@ -426,45 +426,7 @@ void TTabCom::RehashAll()
 
 const TSeqCollection *TTabCom::GetListOfClasses()
 {
-   if (!fpClasses) {
-      fpClasses = new THashList;
-      // Iterate over the table from the map file.
-      THashList* entries = gInterpreter->GetMapfile()->GetTable();
-      TIter next(entries);
-      while (const auto key = next()) {
-         // This is not needed with the new rootmap format
-         const char* className = key->GetName();
-         if (!strncmp(className, "Library.", 8))
-            className += 8;
-
-         if (!strstr(className, ".h"))
-            fpClasses->Add(new TObjString(className));
-      }
-
-      // We might have autoload entries that don't have a rootmap entry
-      // (libCore) and no interpreter info (not yet loaded).
-      TClassTable::Init(); // reset counter
-      while (const char* className = TClassTable::Next()) {
-         if (!fpClasses->FindObject(className)) {
-            fpClasses->Add(new TObjString(className));
-         }
-      }
-   }
-
-   if (fPrevInterpMarker != gInterpreter->GetInterpreterStateMarker()) {
-      ClassInfo_t* ci = gInterpreter->ClassInfo_Factory(kFALSE /*all*/);
-      while (gInterpreter->ClassInfo_Next(ci)) {
-         const char* className = gInterpreter->ClassInfo_FullName(ci);
-         if (strstr(className, "(anonymous)"))
-            continue;
-         if (!fpClasses->FindObject(className)) {
-            fpClasses->Add(new TObjString(className));
-         }
-      }
-      gInterpreter->ClassInfo_Delete(ci);
-   }
-
-   return fpClasses;
+   return (TSeqCollection*) gROOT->GetListOfClasses();
```

Comment:
That is far from identical. As an example, `gROOT->GetListOfClasses()` only contains `TClass`es, and those are lazily created.

In general, `TTabCom` wants classes from three sources:
- TClasses (i.e. `gROOT->GetListOfClasses()`)
- rootmap (`GetMapfile()`, things that we could load the library for and create `TClass`es for)
- `TClassTable` (things we have loaded the library for but didn't create a `TClass` for yet)
- types not known to `TClass` (yet) but only to the interpreter, e.g. from a `class MyClass;` at the prompt or in an interpreted script.

The latter is the reason why we iterate over the AST.